dvc/repo/diff.py

```diff
@@ -101,17 +123,25 @@ def _to_checksum(output):
     for stage in repo.stages:
         for output in stage.outs:
             if _exists(output):
-                yield _to_path(output), _to_checksum(output)
-                if output.is_dir_checksum:
-                    yield from _dir_output_paths(repo_tree, output)
+                yield_output = no_targets or output.path_info.isin_or_eq(
+                    targets
+                )
```

Comment:
As mentioned above, this can be done using `any()` without requiring the change to `isin_or_eq`.

Reply:
Incorporated in 3951673.